lib/jose-jws-verify.js

```diff
@@ -183,7 +185,13 @@ JoseJWS.Verifier.prototype.verify = function () {
     if (keyfinder) {
       key_promises[kid] = keyfinder(kid);
     }
-    promises.push(that.cryptographer.verify(sig.aad, that.payload, sig.signature, key_promises[kid], kid));
+    promises.push(that.cryptographer.verify(sig.aad, that.payload, sig.signature, key_promises[kid], kid)
+      .then(function (vr) {
+        if (vr.verified) {
+          vr.payload = Utils.Base64Url.decode(that.payload);
+        }
```

Comment:
Throwing an exception would be a breaking change, for one thing. I also think that it doesn't represent the spirit of the function. We are returning an array of verification results, and perhaps some of them can be verified but others cannot. I would not want to have one failure poison the others, so I think the boolean in the object representing verification status is a better way to indicate what has happened, allowing client code to decide what to do from there without having to worry about wrapping things with exception handlers.